app/resonant-laboratory/general_purpose/binUtils.js

```diff
@@ -19,7 +19,11 @@ function coerceValue (value, coerceToType) {
   } else if (coerceToType === 'number') {
     value = parseFloat(value);
   } else if (coerceToType === 'string') {
-    value = String(value);
+    if (value && value.str) {
+      value = value.str;
```

Comment:
Just curious - what objects have a `str` field like this? I thought `ObjectId`s got turned into JSON with a specially structured `_id` field?